tests/jballerina-unit-test/src/test/resources/test-src/services/service_decl_distinct_listener_arg.bal

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+public type ServiceType distinct service object {
+};
+
+public class Listener {
+    public isolated function 'start() returns error? {
+        return externStart(self);
+    }
+    public isolated function gracefulStop() returns error? {
+    }
+    public isolated function immediateStop() returns error? {
+    }
+    public isolated function detach(ServiceType s) returns error? {
+    }
+    public isolated function attach(ServiceType s, string[]|string? name = ()) returns error? {
+        return self.register(s, name);
+    }
+    isolated function register(service object {} s, string[]|string? name) returns error? {
+        return externAttach(s, name);
+    }
+
+    public function init() returns error? {
+        check externLInit(self);
+    }
+}
+
+isolated function externAttach(service object {} s, string[]|string? name) returns error? = @java:Method {
+    'class: "org/ballerinalang/nativeimpl/jvm/servicetests/ServiceValue",
+    name: "attach"
+} external;
+
+isolated function externStart(object {} o) returns error? = @java:Method {
+    'class: "org/ballerinalang/nativeimpl/jvm/servicetests/ServiceValue",
+    name: "start"
+} external;
+
+isolated function externLInit(object {} o) returns error? = @java:Method {
+    'class: "org/ballerinalang/nativeimpl/jvm/servicetests/ServiceValue",
+    name: "listenerInit"
+} external;
+
+function getService() returns object {} = @java:Method {
+    'class: "org/ballerinalang/nativeimpl/jvm/servicetests/ServiceValue",
+    name: "getService"
+} external;
+
+function reset() = @java:Method {
+    'class: "org/ballerinalang/nativeimpl/jvm/servicetests/ServiceValue",
+    name: "reset"
+} external;
+
+public function callMethod(service object {} s, string name) returns future<any|error>  = @java:Method {
+    'class:"org/ballerinalang/nativeimpl/jvm/servicetests/ServiceValue",
+    name:"callMethod"
+} external;
+
+listener lsn = new Listener();
+
+service / on lsn {
+
+    function init() returns error? {
+    }
+
+    resource function get processRequest() returns json {
+        return { output: "Hello" };
+    }
+
+    function createError() returns @tainted error? {
```

Comment:
Out of curiosity, do we still use `@tainted` ?

Reply:
It is not needed. Should be a copied code.